### PR TITLE
Added example code to help users more

### PR DESCRIPTION
**Pull request by seccubus**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -180,6 +180,47 @@ Providers
...
+    # Generate a self-signed if we don't have a cert to prevent bootstrap problems
+    letsencrypt_selfsigned "#{site}" do
+        crt     "/etc/httpd/ssl/#{site}.crt"
+        key     "/etc/httpd/ssl/#{site}.key"
+        chain    "/etc/httpd/ssl/#{site}.pem"
+        owner   "apache"
+        group   "apache"
+        notifies :restart, "service[apache2]", :immediate
+        not_if do
+            # Only generate a slef-signed cert if needed
```
Comment (thoutenbos) [inline README.md]: Typo ?

File: README.md [hunk 2]
```diff
@@ -180,6 +180,47 @@ Providers
...
+
+    # Set up contact information. Note the mailto: notation
+    node.set['letsencrypt']['contact'] = [ 'mailto:me@example.com' ] 
+    # Real certificates please...
+    node.set['letsencrypt']['endpoint'] = 'https://acme-v01.api.letsencrypt.org' 
+
+    site="example.com"
+    sans=Array[ "www.#{site}" ]
+
+    # Set up your server here...
```
Comment (thoutenbos) [inline README.md]: Setup of the webserver should happen after calling the `selfsigned` but before the `certificate` provider

File: README.md [hunk 3]
```diff
@@ -196,3 +237,5 @@ Contributing
 License and Authors
 -------------------
 Authors: Thijs Houtenbos <thoutenbos@schubergphilis.com>
+Contributors:
```
Comment (thoutenbos) [inline README.md]: Formatting here is incorrect

File: README.md (hunk 1 above)
Comment (seccubus) [inline README.md]: Fixed. I'll add a commit to this one.

File: README.md (hunk 2 above)
Comment (seccubus) [inline README.md]: I tried that first what I found out is taht I had to set up the web server configuraiton first, then create the self signed certificate and call an immediate restart after that to make sure that the webserver is actually running. You could do it the other way around, but then you would have to call an immediate restart every time your web server setup changes just to make sure the web server is running so I can renew/request the LetsEncrypt certificate.

**Comment by seccubus** [comment]
Please merge @thoutenbos

**Pull request merged**